pkg/kubelet/allocation/allocation_manager_test.go

```diff
@@ -173,8 +197,23 @@ func TestUpdatePodFromAllocation(t *testing.T) {
 		},
 		expectUpdate: true,
 		expectPod:    resizedPod,
+	}, {
+		name: "with resized pod-level resource allocation",
+		pod:  pod,
+		allocated: state.PodResourceInfo{
+			ContainerResources: map[string]v1.ResourceRequirements{
+				"c1":                  *resizedPod.Spec.Containers[0].Resources.DeepCopy(),
+				"c2":                  *resizedPod.Spec.Containers[1].Resources.DeepCopy(),
+				"c1-restartable-init": *resizedPod.Spec.InitContainers[0].Resources.DeepCopy(),
+				"c1-init":             *resizedPod.Spec.InitContainers[1].Resources.DeepCopy(),
+			},
+			PodLevelResources: resizedPodWithPodLevelResources.Spec.Resources.DeepCopy(),
+		},
+		expectUpdate: true,
+		expectPod:    resizedPodWithPodLevelResources,
 	}}
 
+	featuregatetesting.SetFeatureGateDuringTest(t, utilfeature.DefaultFeatureGate, features.InPlacePodLevelResourcesVerticalScaling, true)
```

Comment:
General comment: This file needs more updates; we need to modify TestRetryPendingResizes, TestAllocationManagerAddPod, TestIsResizeIncreasingRequests, and TestSortPendingResizes to include coverage for pod-level resources